httputil/httputil.go

```diff
@@ -16,12 +17,24 @@ type Consumer interface {
 	String() string
 }
 
+type stateMachine interface {
+	State() (*statemachine.State, time.Time)
+}
+
+type Task struct {
+	ID       string     `json:"id"`
+	Started  time.Time  `json:"started"`
+	Stopped  *time.Time `json:"stopped,omitempty"`
+	State    string     `json:"state,omitempty"`
+	Modified *time.Time `json:"modified,omitempty"`
```

Comment:
[nit] instead of a pointer/omitempty should it be required?  starting out with the creation time?